### ISSUE TITLE
Don't apply markdown formatting to YAML Front Matter in preview

### ISSUE THREAD

**Issue by sdaugherty** [HoverBaum/BlankUp-Electron#3]
If the first line of the file is `---`, it's very likely that the file contains YAML Front Matter, and everything until the 2nd `---` should either be ignored entirely, or perhaps rendered as a table like Github's Markdown preview does.

I realize this is a small issue, but it's ugly and distracting.

**Comment by HoverBaum** [comment]
Hi,

thanks for the feedback. Sounds like a good idea. I am looking to add a few features to support people writing blogs (I guess that is your use case). This could be a good first step.

**Comment by sdaugherty** [comment]
Indeed, blogging (and other static site applications) are my use cases. 
I have quite a few Markdown documents over at https://github.com/irchelp/wio that I'm maintaining now, and I'm always looking for better tools. :)

**Comment by HoverBaum** [comment]
Blogging and site generation is also what I mainly want to use this tool for. Rather midterm than shortterm I want to support features for that, like frontmatter editing. Not quite sure how that will look and if it will be a plugin based system or a differnt program or a mode you can activate.

So if you have any feature suggestions I would be open for that.

Also would you like to see the frontmatter hidden or just highlighted differently? And could you link me a reference for the gitHub preview, not quite sure what you are referring to.
